dspback/triggers.py

```diff
@@ -58,24 +61,28 @@ async def watch_discovery_with_retry():
 
 
 async def watch_submissions():
+    logger.info(f"Starting watching Submissions")
     db = motor.motor_asyncio.AsyncIOMotorClient(get_settings().mongo_url)[get_settings().mongo_database]
     async with db["Submission"].watch(
         full_document="updateLookup", full_document_before_change="whenAvailable"
     ) as stream:
         async for change in stream:
-            logger.warning(f"start with a {change}")
+            logger.debug(f"processing submission watch for document: {change}")
             if change["operationType"] != "delete":
                 document = change["fullDocument"]
                 public_json_ld = await retrieve_submission_json_ld(document)
 
                 if public_json_ld:
+                    logger.debug(f"Found public jsonld, updating the discovery record for {document['identifier']}")
                     await db["discovery"].find_one_and_replace(
                         {"repository_identifier": public_json_ld["repository_identifier"]}, public_json_ld, upsert=True
                     )
                 else:
+                    logger.debug(f"No public jsonld found, deleting the discovery record for {document['identifier']}")
                     result = await db["discovery"].delete_one({"repository_identifier": document["identifier"]})
                     logger.warning(f"delete count {result.deleted_count}")
             else:
+                logger.debug(f"Deleting the discovery record for {document['identifier']}")
                 document = change["fullDocumentBeforeChange"]
```

Comment:
Need to go before the logging statement in order to have access to the `document`?

Reply:
nice catch, fixed